calcOutdatedMetrics.js

```diff
@@ -1,69 +1,80 @@
+const { exec } = require("child_process");
+
 process.stdin.resume();
-process.stdin.setEncoding('utf8');
-let data = '';
+process.stdin.setEncoding("utf8");
+let data = "";
 
-process.stdin.on('data', function (chunk) {
+process.stdin.on("data", function (chunk) {
   data += chunk;
 });
 
-process.stdin.on('end', function () {
-  const jsonLines = data.split('\n');
-  const lineWithDependencyTable = jsonLines.find(json =>
-    json.includes('{"type":"table"'),
-  );
+process.stdin.on("end", function () {
+  exec(
+    "cat package.json | jq '.dependencies, .devDependencies | keys | length' | paste -s -d+ - | bc",
+    (err, stdout) => {
+      const jsonLines = data.split("\n");
+      const lineWithDependencyTable = jsonLines.find((json) =>
+        json.includes('{"type":"table"')
+      );
 
-  if (!lineWithDependencyTable) {
-    console.log('data:', data);
-    throw new Error(
-      `Did not receive output from 'yarn outdated'. data: ${data}`,
-    );
-  }
-  const parsedData = JSON.parse(lineWithDependencyTable);
-  const packageList = parsedData.data.body;
+      if (!lineWithDependencyTable) {
+        console.log("data:", data);
+        throw new Error(
+          `Did not receive output from 'yarn outdated'. data: ${data}`
+        );
+      }
+      const parsedData = JSON.parse(lineWithDependencyTable);
+      const packageList = parsedData.data.body;
 
-  const numberOfPackages = packageList?.length;
+      const numberOfPackages = parseInt(stdout);
 
-  const upToDate = [];
-  const behindPatch = [];
-  const behindMinor = [];
-  const behindMajor = [];
+      const behindPatch = [];
+      const behindMinor = [];
+      const behindMajor = [];
 
-  packageList?.forEach(package => {
-    const [name, currentVersion, wanted, latestVersion] = package;
-    const [currentMajor, currentMinor, currentPatch] = currentVersion;
-    const [latestMajor, latestMinor, latestPatch] = latestVersion;
+      packageList?.forEach((package) => {
+        const [name, currentVersion, wanted, latestVersion] = package;
```

Comment:
Love the parallel assignment,
though I got burned recently doing something similar with the output of sequelize update since the size of the return array changes based on what's updated

Do these arrays here guarantee a return size/structure/ordering?

Reply:
This is the output from `yarn outdated --json`. Try running it yourself, with and without the json flag, and that should give you a good idea of what we're looking at.

Here's the output from the shared repo:
```json
{
  "type": "table",
  "data": {
    "head": ["Package", "Current", "Wanted", "Latest", "Package Type", "URL"],
    "body": [
      [
        "@types/jest",
        "27.4.1",
        "27.4.1",
        "27.5.0",
        "devDependencies",
        "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/jest"
      ],
      [
        "@typescript-eslint/eslint-plugin",
        "5.21.0",
        "5.21.0",
        "5.22.0",
        "devDependencies",
        "https://github.com/typescript-eslint/typescript-eslint#readme"
      ],
      [
        "@typescript-eslint/parser",
        "5.21.0",
        "5.21.0",
        "5.22.0",
        "devDependencies",
        "https://github.com/typescript-eslint/typescript-eslint#readme"
      ],
      [
        "faker",
        "5.5.3",
        "5.5.3",
        "6.6.6",
        "dependencies",
        "https://github.com/Marak/Faker.js#readme"
      ],
      [
        "jest",
        "27.5.1",
        "27.5.1",
        "28.0.3",
        "devDependencies",
        "https://jestjs.io/"
      ],
      [
        "jest-mock",
        "27.5.1",
        "27.5.1",
        "28.0.2",
        "dependencies",
        "https://github.com/facebook/jest#readme"
      ],
      [
        "plaid",
        "10.1.0",
        "10.1.0",
        "10.3.0",
        "dependencies",
        "https://github.com/plaid/plaid-node#readme"
      ],
      [
        "ts-jest",
        "27.1.4",
        "27.1.4",
        "28.0.1",
        "devDependencies",
        "https://kulshekhar.github.io/ts-jest"
      ]
    ]
  }
}
```

It looks like `--json` flag does a something of a crude serialization of the table from the default output, which is why it has these ordered arrays instead of something more intuitive like an object for each dependency.

Reply:
oh sweet, i'll try it!